src/features/party/PartiesProvider.tsx

```diff
@@ -115,6 +136,8 @@ const CurrentPartyProvider = ({ children }: PropsWithChildren) => {
   const { data: partyFromQuery, isLoading, error: errorFromQuery } = useCurrentPartyQuery(true);
   const [userHasSelectedParty, setUserHasSelectedParty] = useState(false);
 
+  const { data: roles } = useCurrentPartyRoles(true);
```

Comment:
This should probably also return `isLoading` and be checked below? If we don't really have to wait for this to complete to proceed, I think the whole query could be moved out from here and into (for example) `useExpressionDataSources()`, much like the externalApi functionality does it.

Reply:
Yes, we need to wait for this so adding isLoading :)